### PR TITLE
EASY-1829 & EASY-1794 :  explicit content type & upload response codes

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes
* EASY-1829 - explicit content type header when a response has a body
* EASY-1794 - #86 did not include the original objective of the issue:
  When uploading/deleting files, do not return a file listing

#### When applied it will
* have changes in the swagger documentation: OK -> no-content (when updating an existing file)
* 
* 

#### Where should the reviewer @DANS-KNAW/easy start?

Is the following description adequate for upload of multiple files via either a zip or multiple body parts?
https://github.com/DANS-KNAW/easy-deposit-api/blob/b962a26bdac7b26b6f1def50cff8508f5c6d54e4/docs/api/api.yml#L425-L426

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala [hunk 1]
```diff
@@ -37,14 +37,17 @@ package object servlets extends DebugEnhancedLogging {
   private val post = "-compress(ed)?"
   private val contentTypeZipPattern = s"application/(($pre($post)?)|(x$post))"
 
+  val contentTypeJson: (String, String) = "content-type" -> "application/json;charset=UTF-8"
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: although this 'hardcoded' way seems fine, I think there is a method in Scalatra that does this. Something like `contentType = "..."`. Have you tried using that one? Or did you explicitly choose for this way?

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala [hunk 2]
```diff
@@ -178,25 +179,25 @@ class DepositServlet(app: EasyDepositApiApp)
...
     case e: InvalidDocumentException => badDocResponse(e)
-    case e: ConcurrentUploadException => Conflict(e.getMessage)
-    case e: FileAlreadyExistsException => Conflict("Conflict. The following file(s) already exist on the server: "+e.getMessage)
-    case e: InvalidDoiException => BadRequest(e.getMessage)
-    case e: BadRequestException => BadRequest(e.getMessage)
-    case e: ZipMustBeOnlyFileException => BadRequest(e.getMessage)
-    case e: NotImplementedException => NotImplemented(e.getMessage)
-    case e: SchemaNotAvailableException => InternalServerError(e.getMessage)
+    case e: ConcurrentUploadException => Conflict(e.getMessage, Map(contentTypePlainText))
+    case e: FileAlreadyExistsException => Conflict("Conflict. The following file(s) already exist on the server: " + e.getMessage, Map(contentTypePlainText))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala]: Not sure if this is useful, but you can use interpolation here.

```suggestion
    case e: FileAlreadyExistsException => Conflict(s"Conflict. The following file(s) already exist on the server: ${ e.getMessage }", Map(contentTypePlainText))
```

Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala]: matter of taste

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala (hunk 1 above)
Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: Current usage of this hard-coded constant:
https://github.com/DANS-KNAW/easy-deposit-api/blob/ef76d3475e8bb8c1b9f46453abce6addffdf90a7/src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala#L75

Scalatra's convenience field is not convenient at all in this code pattern of this specific servlet. See the drawbacks of setting it with `before` in rejected PR #92

**Pull request merged**